### DISCUSSION TITLE
Setting up Allied Vision Tech camera

### DISCUSSION THREAD

**Discussion by theo2261** [bonsai-rx/bonsai#935] [Q&A]
Hello,

We are trying to set up our AVT camera - we've installed the Allied Vision package for Bonsai, then tried to start Bonsai from the command line using the --noboot flag as instructed on other forums, but when trying this the initial Bonsai start page opens, then when I select 'new project' or 'open project' this window closes, and no Bonsai window opens. 
We've also tried installing the AVT Direct Stream Driver to try and get the camera to be recognised by a VideoCaptureDevice node, but the installer doesn't recognise our camera - "Allied Vision Alvium 1800 U-158m". At the moment all we'd like to do is be able to view the camera feed on Bonsai.
Does anyone have any ideas?
Thanks

**Comment by theo2261** [comment]
Have now sorted the --noboot problem by directly opening an already saved Workflow in the command line, just getting a "camera.open failed" error or "feature.set.enum failed". We're using the vimba SDK driver

**Reply by inchinn1** [reply]
We are still struggling to get our Allied Vision Alvium 1800 U-158m to work with Bonsai. We have the latest version of Vimba SDK installed and the Bonsai Allied Vision Library. We can create a VimbaCapture node but when trying to run the workflow we initially get a "feature.set.enum failed" error and then a "camera.open failed" error on any subsequent attempts to run the work flow. When we close Bonsai the cmd widnow has the error in the screen shot.

Any help would be much appreciated.

![Capture](https://user-images.githubusercontent.com/21177232/173851351-d9424899-2cee-4de5-b390-5f7803ef3aa8.PNG)

**Reply by inchinn1** [reply]
Or when trying to run the workflow with the property 'ResetDevice' as True bonsai crashes with the below error, even if bonsai is launched with --noboo
![Capture1](https://user-images.githubusercontent.com/21177232/173853572-699eaf3e-8428-457b-b00a-443b82fa17c8.PNG)
t

**Reply by inchinn1** [reply]
We still haven't managed to get this to work, any help or suggestions would be appreciated

**Reply by inchinn1** [reply]
Currently, both point gray and basler have significant delays in being able to ship new cameras. It seems only allied vision are able to ship cameras at the moment... it would be great if these were compatible with bonsai.

**Reply by cosmouse** [reply]
We are having a similar issue and just posted on the Bonsai Google group forum:

Hello,

We have an issue while trying to set up our Allied Vision camera (model - Allied Vision 1800 U-240m). We installed the allied vision package (https://github.com/jordan-g/Bonsai.AlliedVision), the Allied Vision SDK, and are using the VimbaCapture node to try to access the camera data.

When trying to run the application this way we are getting a Runtime Error, "Camera.Open failed." The camera works fine with the native Vimba software. A similar issue is also mentioned here: https://github.com/bonsai-rx/bonsai/discussions/935. We tried starting Bonsai with the --noboot flag  as mentioned in the package, but this also did not work. The camera also is not accessible through other camera nodes (CameraCapture, VideoCaptureDevice).

We were wondering if one possibility could be the SDK version? For example, perhaps the Bonsai Allied Vision package was designed based on a specific version of the SDK that is deprecated?

Thanks!
Phil

**Reply by patsims** [reply]
Hi all, 
I'm going to jump the wagon on this question because we're having a similar problem, that is, we can't make Bonsai to open our AV camera:

We have a USB3 camera form AV, Mako U-130 B  that works with the Vimba v6.0. In it we saved a *.xml file for camera settings. 
In Bonsai, we downloaded the recent package Vimba library. In the Vimba node we added the path for the camera settings file, and Bonsai is able to grab the camera's  Serial number (FrameCount and Index are empty). 
However, running the workflow will give  "Runtime Error: Camera.LoadCameraSettings failed."
I've tried the --noboot flag suggestions, but with no luck. 
I will be really grateful for any suggestion to solve this problem. 
Thank you so much, 
All the best, 
Patricio

**Reply by cosmouse** [reply]
We have unfortunately put this issue on the back burner for now because we ran into some roadblocks that seem very difficult to overcome. The "Camera.LoadCameraSettings failed" error can be avoided if you go into the code and remove the line that attempts to set a particular feature related to FPS acquisition mode (this is possibly either an issue because it's read-only on some cameras or the code has potentially changed). Even if you get past that particular error, we could never get past an error with "Cannot pass a GCHandle across AppDomains." This is something that the developer originally ran into and Gonçalo suggested could be related to the fact that they're using a legacy .NET version for the Vimba SDK. As far as we can tell, this package may not work consistently unless Allied Vision makes significant updates to their SDK. You can find the original thread where Jordan and Gonçalo were working through this issue here: https://groups.google.com/g/bonsai-users/c/7h799fF42Dk/m/Unn1xEwIBAAJ. But please update us if you manage to make any progress on this issue!

Thanks,
Phil

**Reply by glopesdev** [reply]
@cosmouse @patsims The previous package was implemented by Jordan in a way that unfortunately made it tied to specific camera settings.

We have recently released a new official package: https://www.nuget.org/packages/Bonsai.Vimba

We will focus our maintenance efforts in this new package. Make sure to use the specified version of Vimba (5.1.0 at time of writing).

Note also that the "--no-boot" option is still required due to unfortunate implementation choices by AVT which are beyond our control.

**Reply by patsims** [reply]
Ola @glopesdev and @cosmouse,  
Thank you for your reply and info. I tried indeed your recent Bonsai.Vimba package, but unfortunately I only have the last Vimba6.0 version (couldn't find previous versions in AV site). 
Thank you for your efforts.
All the best, 
Patricio

**Reply by cosmouse** [reply]
Thanks Gonçalo! Patricio, if you email Allied Vision support they will provide you with a link to the previous SDK version

**Comment by Z-Ath** [comment]
Hi there, 

I am working with Allied Vision cameras on Vimba 5.1.0, so far successfully via Bonsai on Windows but I am having trouble with my Linux Ubuntu 20.04 system. I followed the Bonsai install instructions (Including the OpenCV 2.4 build) from here (https://neurogears.org/swc-2020/worksheets/installing/). They look slightly different to the instructions on github (https://github.com/orgs/bonsai-rx/discussions/1101) but very similar - one difference is that the latter links to a Open CV 2.4 tarball whereas i used the .zip file linked from neurogears .. but the build appeared to be successful, i.e. no errors. 

I have the Bonsai.Vimba package but it crashes whenever I click on the serial number parameter, the error I get is: [dir of bonsai.vimba pacakge]/VimbaNET.dll' contains native code that cannot be executed by Mono on this platform. The assembly was probably created using C++/CLI. I have tried on Bonsai 2.6. and 2.8. 

I installed Mono following these instructions (https://www.mono-project.com/download/stable/#download-lin) but using the mono-complete in attempt to cover all bases. 

Any help or suggestions will be appreciated. I noticed on this thread that working with the AV/Bonsai/Linux combination has been very problematic - is that still the case? 

Thanks!! Zoe :)

**Reply by bruno-f-cruz** [reply]
VimbaNET has native dependencies that are locked to the OS you are working with. The current package will likely not work with linux since it was only compiled against Windows DLLs. Feel free to open an issue in the Vimba package or try to compile them yourself and let us know how it goes!

**Reply by Z-Ath** [reply]
Thanks for your quick response - on the off chance someone has used it with linux i did post an issue in the vimba package. Do you know if people have used the pylon package in linux for streaming basler cameras?
